test/e2e/connection.spec.ts

```diff
@@ -29,26 +29,14 @@ test.describe('MSSQL Extension - Database Connection', async () => {
 		const profileName = getProfileName();
 		await addDatabaseConnection(vsCodePage, serverName, databaseName, authType, userName, password, savePassword, profileName);
 
-		let addedSqlConnection: Locator;
-		if (profileName) {
-			addedSqlConnection = await vsCodePage.locator(`div[aria-label="${profileName}"]`);
-		}
-		else {
-			addedSqlConnection = await vsCodePage.getByText(`${serverName}`);
-		}
-
-		await expect(addedSqlConnection).toBeVisible({ timeout: 20 * 1000 });
-
-		await addedSqlConnection.click({ button: 'right' });
-		const disconnectOption = await vsCodePage.locator('span[aria-label="Disconnect"]');
-		await disconnectOption.click();
-		const isDiconnectOptionVisible = await disconnectOption.isVisible()
-		if (isDiconnectOptionVisible) {
-			await disconnectOption.click();
-		}
-
-		await addedSqlConnection.click({ button: 'right' });
-		await expect(disconnectOption).toBeHidden({ timeout: 10000 });
+		await openNewQueryEditor(vsCodePage, profileName, password);
+		await new Promise(r => setTimeout(r, 2 * 1000));
```

Comment:
Hardcoded timeouts like this aren't great since they tend to be kind of flakey - could you not wait for a specific element to be visible/hidden with a reasonable timeout instead? (e.g. what you did for the disconnected status below)